### ISSUE TITLE
Allow to select specific changlelist on file operations

### ISSUE THREAD

**Issue by evg656e** [mjcrouch/vscode-perforce#185]
If there is more than one changelist in the workspace, allow to select the desired changelist, like it is done in p4v.

![select_pending_changelist](https://user-images.githubusercontent.com/28028005/93330916-316f9d00-f828-11ea-94a9-c0364a620d04.png)

**Comment by mjcrouch** [comment]
Hi,

Thanks for the suggestion, this is something I've thought about previously.

I think choosing the changelist as you perform the action might be a little difficult to implement in a consistent way that a user would understand. For example if you have the extension set to automatically open a file every time you edit it, then we would have to either prompt the user on every save or just always open in the default changelist on save, which is not very consistent. That's not to rule this approach out - It might not be as confusing as I think.

IIRC the jetbrains perforce plugin and maybe eclipse or visual studio plugins have something like a "target" changelist that you choose in advance, so whenever you open or add a file it puts it in to that changelist by default. This might be easier to implement but I think it is still challenging to come up with a consistent behaviour that works without weird errors. - particularly if you also use an external perforce client to perform actions (e.g. your target changelist might be submitted externally).

So to summarise, I think it could be a useful addition, but needs some careful thought and may take some time to implement

**Comment by evg656e** [comment]
>For example if you have the extension set to automatically open a file every time you edit it, then we would have to either prompt the user on every save or just always open in the default changelist on save, which is not very consistent.

imo it would be fine to show prompt only when file editing is invoked from vs menu (otherwise files should go into default changelist, at least this is how the official client works).

>IIRC the jetbrains perforce plugin and maybe eclipse or visual studio plugins have something like a "target" changelist that you choose in advance, so whenever you open or add a file it puts it in to that changelist by default

it's a good option too (at least in case when you mainly work with one changelist in a time)

**Comment by sandipchitale** [comment]
+1

If we can mark a specific change list/workspace folder, as target change list, that will be great.  I may submit a PR.
